add cvector to libCplus

KaiBelmo opened the pull request:
cvector is dynamic array implementation (basic data structure)

KaiBelmo committed:
add cvector to lib9wada

KernelOverseer commented:
I'll review the code tomorrow, thanks for your contribution

KernelOverseer commented on includes/cvector.h:
```diff
@@ -0,0 +1,24 @@
+    int size;
+}cvector;
+
+cvector *create(void);
```
I think function names should change to have a prefix, like `cvector_create` and `cvector_push_back` so there is no conflict with similar named functions in the future.

KernelOverseer commented on sources/cvector/cvector.c:
```diff
@@ -0,0 +1,78 @@
+
+cvector *create(void){
+    cvector *vec = (cvector*)malloc(sizeof(cvector));    
+    vec->data = malloc(sizeof(void*) * INITIAL_SIZE);
```
A good practice is to check for malloc's return.
So when the allocation fails, the element just does not get created,
instead of having a bus error trying to access `NULL` address.

KernelOverseer commented on sources/cvector/cvector.c:
```diff
@@ -0,0 +1,78 @@
+    vec->data = malloc(sizeof(void*) * INITIAL_SIZE);
+
+    memset(vec->data, 0, (INITIAL_SIZE * sizeof(vec->data)));
+    if ((vec == NULL) || (vec->data == NULL))
```
I see that malloc's return here is checked, but don't you think the line before with memset can cause a problem ?

KernelOverseer commented on sources/cvector/cvector.c:
```diff
@@ -0,0 +1,78 @@
+    *magic*
+*/
+
+void delete(cvector *vec){
```
i suggest having a more versatile version of delete that takes a function pointer,
So if it's a list of some struct, you can pass a function that frees all the struct's fields correctly.

KernelOverseer reviewed (changes requested):
Nice implementation and very helpful module,
Just some small changes following some best practices will make it better.

KernelOverseer commented on sources/cvector/cvector.c:
```diff
@@ -0,0 +1,78 @@
+    if ((vec == NULL) || (vec->data == NULL))
+    {
+        perror("malloc");
+        exit(-1);
```
A program shouldn't always exit after malloc fails,
it's not up to this function to decide, it should only return NULL,
then the developer will take the decision depending on the usage.

KaiBelmo committed:
change functions name

KaiBelmo committed:
check if malloc fails

KernelOverseer commented:
i'll do some other changes and then merge

KaiBelmo committed:
update readme

KernelOverseer committed:
Done necessary changes
added cvector_init function to initialise cvector allocated in stack
added a del function as argument to cvector_delete
added a new function cvector_resize to set a wanted size for the cvector

KernelOverseer committed:
Merged with collin's latest commit
Updated the README.md in the module directory
Updated the README.md in the sources directory

KernelOverseer reviewed (approved):
All good now, tested wverything is working.thanks @CollinBelmo for the contribution :heart:
